### DISCUSSION TITLE
Trace analyser

### DISCUSSION THREAD

**Discussion by marcnol** [marcnol/pyHiM#115] [Announcements]
This script in `/src/postProcessing)/trace_analyzer.py` will read a trace table and derive several different analysis, including the distribution of number barcodes per trace: total, duplicated, and non-duplicated. It also exports the Rg distribution.

To analyse a single or a list of unrelated files:
```
$ ls buildsPWDmatrix/Trace_2D_barcode_mask:mask0_ROI:2.ecsv | trace_analyzer.py --pipe
```

To analyze all trace files in `buildPWDmatrix`:

```
$  trace_analyzer.py
```

Example outputs:

![image](https://user-images.githubusercontent.com/341757/215075387-4ed67742-b515-4137-95a9-ddc154ce9454.png)

**Comment by jbfiche** [comment]
Would it be more useful to have it like this?

![Trace_stat](https://user-images.githubusercontent.com/40423104/215157551-a282fd14-b43b-48a5-9de0-768a595e5ef2.png)

That way you can check:
- the number of detections for each barcode (and whether they are duplicated or not)
- how long are the traces (efficiency = unique_barcode / total_barcode) and how many of them are presenting duplicated barcodes
- dropout are all the traces with only one barcode.

**Comment by marcnol** [comment]
Yes this is very helpful. Do u want to do a PR?

On Fri 27 Jan 2023 at 18:49, Fiche ***@***.***> wrote:

> Would it be more useful to have it like this?
>
> [image: Trace_stat]
> <https://user-images.githubusercontent.com/40423104/215157551-a282fd14-b43b-48a5-9de0-768a595e5ef2.png>
>
> That way you can check:
>
>    - the number of detections for each barcode (and whether they are
>    duplicated or not)
>    - how long are the traces (efficiency = unique_barcode /
>    total_barcode) and how many of them are presenting duplicated barcodes
>    - dropout are all the traces with only one barcode.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/marcnol/pyHiM/discussions/115#discussioncomment-4799467>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACTN7O3O6Z2AKEIUMSBGODWUQDDHANCNFSM6AAAAAAUISX5H4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 
||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
Marcelo Nollmann
CNRS Research Director
e-mail: ***@***.***, ***@***.***
website: http://www.nollmannlab.org/

Centre de Biochimie Structurale, INSERM/CNRS
GPS coordinates: 43.636141, 3.845820
Rue Serge Reggiani
34090 Montpellier
France
 ________________________________________
/ Caution: breathing may be hazardous to \
\ your health.                                                   /
 -----------------------------------------------------------------
   o
    o              ....
           ........    .
          .            .
         .             .
.........              .......
..............................
Elephant inside ASCII snake
fortune -a | fmt -80 -s | $(shuf -n 1 -e cowsay cowthink) -$(shuf -n 1 -e b
d g
p s t w y) -f $(shuf -n 1 -e $(cowsay -l | tail -n +2)) -n

**Reply by marcnol** [reply]
I added histograms of xyz localizations.

**Comment by marcnol** [comment]
Example:

```
$ trace_analyzer.py --input Trace_3D_barcode_mask:DAPI_ROI:1_filtered.ecsv
```

will analyze the trace file and produce several outputs, including:


![Trace_3D_barcode_mask:DAPI_ROI:3_filtered_traces_XYZ_ROI3](https://github.com/marcnol/pyHiM/assets/341757/2b3f32f2-d9a6-41c8-98b7-372cc60a0439)

![Trace_3D_barcode_mask:DAPI_ROI:3_filtered_trace_statistics](https://github.com/marcnol/pyHiM/assets/341757/281cf895-d043-422c-a7c1-5fc7dcbbf857)

![Trace_3D_barcode_mask:DAPI_ROI:3_filtered_xyz_statistics](https://github.com/marcnol/pyHiM/assets/341757/c33e6a4b-0678-4f1e-b1cd-8834e0779560)
